### ISSUE TITLE
Changing configuration of scanners after initializing with loading models

### ISSUE THREAD

**Issue by nashugame** [protectai/llm-guard#154]
Hi Guys, I am trying to LLM guard in our group enterprise's solution. The problem we are facing we want to make different paramters of input scanners to changed as per user request dynamically, one way to do it will reinitialize scanner on each user request which will reload the models (associated with different scanner config). But this is not going to scale up. 
I am looking at something wherein we can initialize scanner only once (load model once with default config) and dynamically change the config of these scanners as per user's request. I want to know if it's at all possible to achieve this  and how?

**Comment by asofter** [comment]
Hey @nashugame , thanks for reaching out. We have received such requests multiple times. However, it will require refactoring of all scanners which we are planning eventually but it's not a priority at a given moment. Yet, we'd be happy to accept Pull Requests with suggestions on how to achieve this.

**Comment by toffentoffen** [comment]
@asofter What would need to be done, how do you envision the refactor? Me and others could work on it.
